iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -640,7 +640,13 @@ public void preTruncateTable(org.apache.hadoop.hive.metastore.api.Table table, E
       throws MetaException {
     this.catalogProperties = getCatalogProperties(table);
     this.icebergTable = Catalogs.loadTable(conf, catalogProperties);
-    Map<String, PartitionField> partitionFieldMap = icebergTable.spec().fields().stream()
+    truncatePartitionBySpec(icebergTable, partNames, icebergTable.newDelete());
+    context.putToProperties("truncateSkipDataDeletion", "true");
```

Comment:
could we please use `TRUNCATE_SKIP_DATA_DELETION` constant here.

Reply:
Changed to the constant. I will check if that is used in case of Iceberg.

Reply:
i don't see constant used here